sources/packages/api/src/route-controllers/confirmation-of-enrollment/confirmation-of-enrollment.controller.ts

```diff
@@ -191,53 +228,57 @@ export class ConfirmationOfEnrollmentController {
    */
   @HasLocationAccess("locationId")
   @Patch(
-    ":locationId/confirmation-of-enrollment/application/:applicationId/confirm",
+    ":locationId/confirmation-of-enrollment/disbursement/:disbursementScheduleId/confirm",
   )
   async confirmEnrollment(
     @Param("locationId") locationId: number,
-    @Param("applicationId") applicationId: number,
+    @Param("disbursementScheduleId") disbursementScheduleId: number,
+    @UserToken() userToken: IUserToken,
   ): Promise<void> {
-    // get application for the location if COE status is `REQUIRED`
-    const application =
-      await this.applicationService.getApplicationDetailsForCOE(
+    // Get the disbursement and application summary for COE.
+    const disbursementSchedule =
+      await this.disbursementScheduleService.getDisbursementAndApplicationSummary(
         locationId,
-        applicationId,
+        disbursementScheduleId,
       );
-    if (!application) {
-      throw new NotFoundException("Application Not Found");
+
+    if (!disbursementSchedule) {
+      throw new NotFoundException(COE_NOT_FOUND_MESSAGE);
     }
     // institution user can only confirm COE, when the student is
-    // within COE_WINDOW of their Program Start date
+    // within COE_WINDOW of disbursement date
     if (
       !this.applicationService.withinValidCOEWindow(
-        application.offering.studyStartDate,
+        disbursementSchedule.disbursementDate,
       )
     ) {
       throw new UnprocessableEntityException(
         `Confirmation of Enrollment window is greater than ${COE_WINDOW} days`,
       );
     }
 
-    //TODO: Document Number should be Updated in the same transaction when COE status changes
-    await this.disbursementScheduleService.updateDisbursementScheduleDocumentNumber(
-      applicationId,
-    );
-
-    const updatedCOEStatus =
-      await this.applicationService.updateApplicationCOEStatus(
-        applicationId,
-        COEStatus.completed,
-        ApplicationStatus.completed,
+    const firstOutstandingDisbursement =
+      await this.disbursementScheduleService.getFirstOutstandingCOE(
+        disbursementSchedule.application.id,
       );
 
-    if (updatedCOEStatus.affected === 0) {
+    if (disbursementSchedule.id !== firstOutstandingDisbursement.id) {
       throw new UnprocessableEntityException(
-        "Confirmation of Enrollment and application status update to completed is failed",
+        `${FIRST_COE_NOT_COMPLETE} ${FIRST_COE_NOT_COMPLETE_MESSAGE}`,
```

Comment:
Can you please elaborate what is the idea behind concatenation these two strings?

Reply:
I saw now why it was done but I also share the other way of doing the same thing that was already in place for a while. Please check further down in the comments and we can talk about it.

Reply:
First one is the error code and next is the error message concatenated.

The idea is, for errors with error code, the custom error message should be displayed on toast instead of a generic error message. 

![image](https://user-images.githubusercontent.com/54600590/153092113-4c124e5b-ccba-47ec-951d-17a0fca4b3be.png)

Reply:
As discussed, using error type I am setting the custom message at the Vue side and throwing ApiProcessError from API side.